src/software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,145 @@
+#include "software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_fsm.h"
+
+#include "proto/parameters.pb.h"
+#include "shared/constants.h"
+#include "software/ai/evaluation/defender_assignment.h"
+#include "software/ai/evaluation/enemy_threat.h"
+#include "software/ai/hl/stp/tactic/crease_defender/crease_defender_tactic.h"
+#include "software/ai/hl/stp/tactic/move/move_tactic.h"
+#include "software/ai/hl/stp/tactic/pass_defender/pass_defender_tactic.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/util/generic_factory/generic_factory.h"
+
+EnemyFreeKickPlayFSM::EnemyFreeKickPlayFSM(TbotsProto::AiConfig ai_config)
+        : DefensePlayBase::DefensePlayBase(ai_config)
+{
+}
+
+void EnemyFreeKickPlayFSM::blockEnemyKicker(const Update& event)
+{
+    setTactics(event, event.common.num_tactics);
+}
+
+void EnemyFreeKickPlayFSM::setTactics(const Update& event, unsigned int num_tactics)
+{
+    // One tactic is always designated as the free kick defender
+    unsigned int num_defenders             = num_tactics - 1;
```

Comment:
let's use an int here. Suppose `num_tactics` is 0, then `num_defenders` would underflow in this calculation to 2^32,

Reply:
In the case it is 0, the shouldn't we also not assign a free kick defender? So it might be better to just add a if statement at the start for `num_tactics >= 1`.